backend/app/store/search/search_test.go

```diff
@@ -0,0 +1,333 @@
+
+	t0 := time.Date(2017, 12, 20, 15, 18, 24, 0, time.Local)
+	for shift := 0; shift < 4; shift++ {
+		cid := fmt.Sprintf("comment%d", shift)
+		searcher.IndexDocument(cid, &store.Comment{
+			ID:        cid,
+			Locator:   store.Locator{SiteID: "test-site", URL: fmt.Sprintf("http://example.com/post%d", shift%2)},
+			Text:      "text 123",
+			User:      store.User{ID: "u1", Name: "user1"},
+			Timestamp: t0.Add(time.Duration(shift) * time.Minute),
+		})
+	}
+
+	_ = searcher.Flush("test-site")
+	{
+		res, err := searcher.Search(&Request{SiteID: "test-site", Query: "123", Limit: 1, From: 0})
+		require.NoError(t, err)
+		require.Len(t, res.Documents, 1)
+		assert.Equal(t, "comment0", res.Documents[0].ID)
+	}
+	{
+		res, err := searcher.Search(&Request{SiteID: "test-site", Query: "123", Limit: 1, From: 1})
+		require.NoError(t, err)
+		require.Len(t, res.Documents, 1)
+		assert.Equal(t, "comment1", res.Documents[0].ID)
+	}
+	{
+		res, err := searcher.Search(&Request{SiteID: "test-site", Query: "123", Limit: 1, From: 3})
+		require.NoError(t, err)
+		require.Len(t, res.Documents, 1)
+		assert.Equal(t, "comment3", res.Documents[0].ID)
+	}
+	{
+		res, err := searcher.Search(&Request{SiteID: "test-site", Query: "123", Limit: 2, From: 1, SortBy: "-timestamp"})
+		require.NoError(t, err)
+		require.Len(t, res.Documents, 2)
+		assert.Equal(t, []string{"comment2", "comment1"}, []string{res.Documents[0].ID, res.Documents[1].ID})
+	}
+}
+
+func createDB(t *testing.T, commentsPerSite int, sites []string) (e engine.Interface, teardown func()) {
+	testDB := os.TempDir() + "/remark-db"
+	_ = os.RemoveAll(testDB)
+	err := os.MkdirAll(testDB, 0700)
+	require.NoError(t, err)
+	bsites := []engine.BoltSite{}
+	for _, s := range sites {
+		bsites = append(bsites, engine.BoltSite{FileName: testDB + "/" + s, SiteID: s})
+	}
+	b, err := engine.NewBoltDB(bolt.Options{}, bsites...)
+	require.NoError(t, err)
+	teardown = func() {
+		require.NoError(t, b.Close())
+		_ = os.RemoveAll(testDB)
+	}
+
+	rng := rand.New(rand.NewSource(42))
+
+	t0 := time.Date(2017, 12, 20, 15, 18, 24, 0, time.Local)
+	for _, siteID := range sites {
+		for shift := 0; shift < commentsPerSite; shift++ {
+			cid := fmt.Sprintf("comment%d", shift)
+			uid := rng.Intn(15)
+			comment := store.Comment{
+				ID:        cid,
+				Locator:   store.Locator{SiteID: siteID, URL: fmt.Sprintf("http://example.com/post%d", rng.Intn(10))},
+				Text:      fmt.Sprintf("%d text %d", rng.Int63(), rng.Int63()),
+				User:      store.User{ID: fmt.Sprintf("u%d", uid), Name: fmt.Sprintf("user %d", uid)},
+				Timestamp: t0.Add(time.Duration(shift) * time.Hour),
+			}
+			ccid, err := b.Create(comment)
+			require.NoError(t, err)
+			require.Equal(t, cid, ccid)
+		}
+	}
+
+	return b, teardown
+}
+
+func TestSearch_IndexStartup(t *testing.T) {
+	sites := []string{"test-site", "remark", "test-site42"}
+
+	searcher, teardown := createTestService(t, sites)
+	defer teardown()
+
+	b, teardown := createDB(t, 42, sites)
```

Comment:
Please rename teaddown functions to unique names as now you have two of them deferred and I am not sure if you won't call the same function (latter one) twice instead of calling two separate functions.